### ISSUE TITLE
Confusing wording and diagnostic range in `D401`

### ISSUE THREAD

**Issue by jonathandung** [astral-sh/ruff#26260]
[Playground](https://play.ruff.rs/715e9db3-7d31-4234-ad0b-2f2f93de0d8a)

**Comment by jonathandung** [comment]
I believe this applies to `D415` as well. Essentially, the terminal output points at where the docstring closes, which is a pain when managing long docstrings since one must navigate to the first line.

**Comment by ntBre** [comment]
Ah yeah, I can see how the caret at the end of the diagnostic range looks like it's pointing to the end of the docstring, but it's just the marker for the end of the range.

```
[D401](https://docs.astral.sh/ruff/rules/non-imperative-mood) First line of docstring should be in imperative mood: "A"
  --> try.py:31:5
   |
29 |   def func4():
30 |
31 | /     '''A
32 | |
33 | |     docstring.''' # This passes, meaning "line" refers to a contingent block of text not containing a blank line
   | |_________________^
```

The start of the range and the diagnostic location (`try.py:31:5`) is still the first line of the docstring, which should help with jumping there in most editors at least.

I thought we intentionally used the full docstring as the range so that you can put a suppression comment at the end of the docstring like:

```py
def func3():
    '''A

    docstring.''' # noqa: D400
```

but that doesn't seem to be the case (https://github.com/astral-sh/ruff/issues/10736#issuecomment-2032088363).

It does seem intentional that we check the first "logical line" rather than literal line (https://github.com/astral-sh/ruff/issues/2994#issuecomment-1435100807), but I agree that the message is confusing in that case.

Overall, I think I agree with you. As long as we can change the diagnostic range without regressing `noqa` behavior, it seems worth doing, as well as clarifying the diagnostic message.
